load.go

```diff
@@ -447,3 +455,75 @@ func (b *ArtifactBuild) processBuildArgs(lex *shell.Lex, args map[string]string,
 
 	return goerrors.Join(errs...)
 }
+
+func (i *ImageConfig) validate() error {
+	if i == nil {
+		return nil
+	}
+
+	var errs []error
+	if err := i.Post.validate(); err != nil {
+		errs = append(errs, errors.Wrap(err, "postinstall"))
+	}
+
+	return goerrors.Join(errs...)
+}
+
+func (p *PostInstall) validate() error {
+	if p == nil {
+		return nil
+	}
+
+	var errs []error
+
+	if err := validateSymlinks(p.Symlinks); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errors.Wrap(goerrors.Join(errs...), "symlink")
+}
+
+func validateSymlinks(symlinks map[string]SymlinkTarget) error {
+	var errs []error
+	dests := make(map[string]string, len(symlinks)<<1)
+
+	for oldpath, cfg := range symlinks {
+		var err error
+		if oldpath == "" {
+			err = fmt.Errorf("symlink source is empty")
+			errs = append(errs, err)
+		}
+
+		if cfg.Path != "" && len(cfg.Paths) != 0 || cfg.Path == "" && len(cfg.Paths) == 0 {
+			err = fmt.Errorf("'path' and 'paths' fields are mutually exclusive, and at least one is required")
+			errs = append(errs, err)
+		}
+
+		if err != nil {
+			continue
+		}
+
+		// By this point, both the oldpath and the SymlinkTarget are
+		// well-formed. We still need to make sure each 'newpath' is unique.
+		if cfg.Paths == nil {
```

Comment:
Validations should not be modifying anything.
We can also iterate on `Paths` without having to have it be non-nil.

Reply:
fixed